lib/ecto/association.ex

```diff
@@ -1049,6 +1045,31 @@ defmodule Ecto.Association.ManyToMany do
     end
   end
 
+  defp validate_join_through(name, nil) do
+    raise ArgumentError, "many_to_many #{inspect name} associations require the :join_through option to be given"
+  end
+  defp validate_join_through(name, %Ecto.Query{} = query) do
```

Comment:
Let's do this validation with pattern matching. I have just posted an example above. :)

Reply:
Done!